transform/app/services/solr/send.py

```diff
@@ -0,0 +1,68 @@
+import json
+import logging
+
+import requests
+from requests.exceptions import ConnectionError as ReqConnectionError
+
+from app.services.solr.utils import get_default_headers
+
+logger = logging.getLogger(__name__)
+
+
+def send_spark_df(
+    json_data: list[str],
+    solr_url: str,
+    collections: list[str],
+    file_key,
+    headers: dict = None,
+) -> None:
+    """
+    Sends JSON data to Solr collections.
+
+    Args:
+        json_data (List[str]): JSON data to be sent.
+        solr_url (str): Solr base URL.
+        collections (List[str]): List of Solr collection names.
+        headers (dict): HTTP headers for Solr request.
+
+    Raises:
+        ReqConnectionError: If there is a connection issue with Solr.
+        Exception: For any unexpected errors during the Solr request.
+    """
+    headers = headers or get_default_headers()
+    json_array = json.dumps([json.loads(line) for line in json_data])
```

Comment:
You have tested that and its good right? 
Before we were doing: https://github.com/cyfronet-fid/eosc-search-service/blob/c805c25fb5c51aae7fd19fe4cc69578b9512ff2c/transform/app/tasks/transform/batch.py#L58

I wonder if that is better/faster way

Reply:
That way we utilise list comprehension instead of using map and lambda. So I think it should be faster. Also I think that we should merge `send_json_string_to_solr` and my `send_json`.